src/GitRepository.cc

```diff
@@ -226,18 +226,18 @@ LibGitCommit GitRepository::get_commit(unsigned int count)
     return parent;
 }
 
-LibGitCommit GitRepository::get_commit(const std::string& ref)
+LibGitCommit GitRepository::get_commit(const std::string& ref) const
 {
     git_commit* commit;
     git_oid oid_parent_commit;
 
     // resolve HEAD into a SHA1
-    int error = git_reference_name_to_id(&oid_parent_commit, repo_.get(), ref.c_str());
+    int error = git_reference_name_to_id(&oid_parent_commit, const_cast<git_repository*>(repo_.get()), ref.c_str());
     if (error)
         throw git::Error{ gul14::cat("Cannot find ID from reference name: ", git_error_last()->message) };
 
     // find commit object by commit ID
-    error = git_commit_lookup(&commit, repo_.get(), &oid_parent_commit);
+    error = git_commit_lookup(&commit, const_cast<git_repository*>(repo_.get()), &oid_parent_commit);
```

Comment:
I am not sure these `const_cast`s are worth doing just for the sake of making some `GitRepository` member functions `const`. After all, they might invoke undefined behavior – we simply don't know what that ugly libgit2 API is doing under the hood.
> $5.2.11/7 - "[Note: Depending on the type of the object, a write operation through the pointer, lvalue or pointer to data member resulting from a const_cast that casts away a const-qualifier68) may produce undefined behavior (7.1.5.1). ]"

Reply:
>  they might invoke undefined behavior

Yes, theoretically.

Whatever, removing them.